### ISSUE TITLE
*.tsx parser fails

### ISSUE THREAD

**Issue by yurii-sorokin** [depcheck/depcheck#201]
Hi. I've tried to use depcheck with tsx files + typescript@2, but with no luck. I reciving parsing error like:
```
"SyntaxError: Unexpectedtoken(5: 16)
atParser.pp$5.raise(.../depcheck/node_modules/babylon/lib/index.js: 4333: 13)
atParser.pp.unexpected(.../depcheck/node_modules/babylon/lib/index.js: 1705: 8)
atParser.pp$3.parseParenAndDistinguishExpression(.../depcheck/node_modules/babylon/lib/index.js: 3776: 10)
atParser.pp$3.parseExprAtom(.../depcheck/node_modules/babylon/lib/index.js: 3629: 19)
atParser.pp$3.parseExprSubscripts(.../depcheck/node_modules/babylon/lib/index.js: 3414: 19)
atParser.pp$3.parseMaybeUnary(.../depcheck/node_modules/babylon/lib/index.js: 3394: 19)
atParser.pp$3.parseExprOps(.../depcheck/node_modules/babylon/lib/index.js: 3324: 19)
atParser.pp$3.parseMaybeConditional(.../depcheck/node_modules/babylon/lib/index.js: 3301: 19)
atParser.pp$3.parseMaybeAssign(.../depcheck/node_modules/babylon/lib/index.js: 3264: 19)
atParser.pp$3.parseExpression(.../depcheck/node_modules/babylon/lib/index.js: 3226: 19)"
``` 

To reproduce go to test/fake_modules/typescript and change component.tsx from

```
import * as React from 'react';

export default class MyComponent extends React.Component<{}, {}> {
}
```
to
```
import * as React from 'react';

export default class MyComponent extends React.Component<{}, {}> {
  render() {
    return (<div>fake</div>)
  }
}
```

and run tests.

You will see that there are 2 failing tests. Both about typescript parser.

**Comment by rajington** [comment]
that PR worked for me, might want to submit a PR

**Comment by lijunle** [comment]
I am sorry that, I don't notice this issue. Why TypeScript compiler not happy with your code? How does it happen?

**Comment by yurii-sorokin** [comment]
As I can see the problem is that `depcheck` uses some default config for compiler options. After PR which added `jsx` option it might work in most cases. But I believe that instead of providing some defaults `depcheck` should try to read `tsconfig` from package first and use it.

As I still can repro the issue and `depcheck` tests are passing I've created the repo.
https://github.com/yurii-sorokin/depcheck-ts-bug

Setup repo:
```bash
npm run bootstrap
```
This command will install all dependencies.

Check that ts code is ok:
```bash
npm run build
```
This command will transpile ts using `tsc`.

Check that depcheck is fail:
```bash
npm run test
```
And finally, this one will run depcheck again to packages (both should have errors).

The result should be like (cutted):
```bash
npm run test                                                                                                                                                             Projects/depcheck-ts-bug master

> depcheck-ts-bug@1.0.0 test ../depcheck-ts-bug
> node scripts/check.js

[CHECK] ../depcheck-ts-bug/packages/pkg-test-one/
[RESULT]
{ dependencies: [ 'react', 'react-materialize', [length]: 2 ],
  devDependencies: [ [length]: 0 ],
  missing: {},
  using: {},
  invalidFiles:
   { '../depcheck-ts-bug/packages/pkg-test-one/src/test-one.tsx':
      { SyntaxError: Unexpected token (7:8)
    at Parser.pp$5.raise (../depcheck-ts-bug/node_modules/babylon/lib/index.js:4454:13)
    ...   
    at Parser.pp$1.parseStatement (../depcheck-ts-bug/node_modules/babylon/lib/index.js:1906:19)',
        [message]: 'Unexpected token (7:8)',
        pos: 234,
        loc: Position { line: 7, column: 8 } } },
  invalidDirs: {} }
[/RESULT]
[/CHECK]

[CHECK] ../depcheck-ts-bug/packages/pkg-test-two/
[RESULT]
{ dependencies: [ 'react', 'react-materialize', [length]: 2 ],
  devDependencies: [ [length]: 0 ],
  missing: {},
  using: {},
  invalidFiles:
   { '../depcheck-ts-bug/packages/pkg-test-two/src/sub-test.two.tsx':
      { SyntaxError: Unexpected token (6:8)
    at Parser.pp$5.raise (../depcheck-ts-bug/node_modules/babylon/lib/index.js:4454:13)
    ...
    at Parser.pp$1.parseStatement (../depcheck-ts-bug/node_modules/babylon/lib/index.js:1906:19)',
        [message]: 'Unexpected token (6:8)',
        pos: 179,
        loc: Position { line: 6, column: 8 } },
     '../depcheck-ts-bug/packages/pkg-test-two/src/test-two.tsx':
      { SyntaxError: Unexpected token (7:8)
    at Parser.pp$5.raise (../depcheck-ts-bug/node_modules/babylon/lib/index.js:4454:13)
    ...
    at Parser.pp$1.parseStatement (../depcheck-ts-bug/node_modules/babylon/lib/index.js:1906:19)',
        [message]: 'Unexpected token (7:8)',
        pos: 226,
        loc: Position { line: 7, column: 8 } } },
  invalidDirs: {} }
[/RESULT]
[/CHECK]
```

**Comment by lijunle** [comment]
> As I can see the problem is that depcheck uses some default config for compiler options. After PR which added jsx option it might work in most cases. But I believe that instead of providing some defaults depcheck should try to read tsconfig from package first and use it.

I see. I suppose that is controlled here:

https://github.com/depcheck/depcheck/blob/153aa50925e4a02f1b418a7d1349bff8f5f925eb/src/parser/typescript.js#L11-L14

We need to change this hard coded compile option to read it from file. Not sure if TypeScript provides util to help here.

@yurii-sorokin Could you please help on this issue?

**Comment by yurii-sorokin** [comment]
I could, I just can't give any estimations.

**Comment by ubnt-marc-khouri** [comment]
@lijunle in my testing, the linked PR (#230) resolves this issue -- is there any chance you would be able to merge it? Thanks!

**Comment by lijunle** [comment]
Hey, @ubnt-marc-khouri I am sorry that, I really don't have time to continue working on this project. If you are interested in fixing these issue, please tell me and I will loop you as contributors.

**Comment by mnkhouri** [comment]
Fixed by #230

**Comment by mnkhouri** [comment]
Thanks @yurii-sorokin !
